### ISSUE TITLE
Weird Segfault.

### ISSUE THREAD

**Issue by crabbo-rave** [StavromulaBeta/cognate#4]
I noticed a weird segfault that appeared in my code. Here is a small reproduction:
This segfaults when ran, yet 
```
Def Push-op as (Let Op; Push Op);
Let Commands be List ("+" "-" "%" "*")
Let Error be False;
List ();
For command in Commands
    Case "+" (Push-op)
    Case "-" (Push-op)
    Case "*" (Push-op)
    else (Set Bad-command; Set Error to True; Prints ("Error: Unknown Command " Bad-command "\n"));
Do If Not Error
    (Set Commands; For command in Commands Print)
    (Print "");
Clear;
```
This code works, 
```
Def Push-op as (Let Op; Push Op);
Let Commands be List ("+" "-" "%" "*");
Let Error be False;
List ();
For command in Commands
    Case "+" (Push-op)
    Case "-" (Push-op)
    Case "*" (Push-op)
    else (Set Error to True; Put "Error: Unknown Command "; Print);
If Not Error
    (Set Commands; For command in Commands Print)
    (Print "");
Clear;
```
and the only difference seems to be this code which causes the segfault
```
else (Set Bad-command; Set Error to True; Prints ("Error: Unknown Command " Bad-command "\n"));
```
as opposed to this code which works.
```
else (Set Error to True; Put "Error: Unknown Command "; Print);
```
Further information: `Prints` doesn't segfault by itself, based on this reproduction I made that works,
```
Let Bar be  "Bar"; Prints ("hielo " Bar "\n");
```

**Comment by StavromulaBeta** [comment]
seems to be segfaulting at compile time

**Comment by StavromulaBeta** [comment]
should be fixed in 86d2fdb
